umh-core/pkg/fsm/redpanda/actions.go

```diff
@@ -155,103 +157,126 @@ func (b *RedpandaInstance) GetServiceStatus(ctx context.Context, filesystemServi
+				return nil
+			}
+			if strings.Contains(err.Error(), redpanda_monitor.ErrServiceConnectionRefused.Error()) {
+				// This is expected during the startup phase of the redpanda service, when the service is not yet ready to receive connections
+				r.baseFSMInstance.GetLogger().Debugf("Service not yet ready: %v", err)
+				return nil
+			}
+			return fmt.Errorf("failed to get observed Redpanda config: %w", err)
+		}
+	})
 
-	wg.Wait()
+	// Create a buffered channel to receive the result from g.Wait().
+	// The channel is buffered so that the goroutine sending on it doesn't block.
+	errc := make(chan error, 1)
 
-	if getServiceStatusErr != nil {
-		return getServiceStatusErr
-	}
+	// Run g.Wait() in a separate goroutine.
+	// This allows us to use a select statement to return early if the context is canceled.
+	go func() {
+		// g.Wait() blocks until all goroutines launched with g.Go() have returned.
+		// It returns the first non-nil error, if any.
+		errc <- g.Wait()
+	}()
 
-	// Store the raw service info
-	b.ObservedState.ServiceInfo = info
-
-	if getConfigErr == nil {
-		// Only update if we successfully got the config
-		b.ObservedState.ObservedRedpandaServiceConfig = observedConfig
-	} else {
-		if strings.Contains(getConfigErr.Error(), redpanda_service.ErrServiceNotExist.Error()) || strings.Contains(getConfigErr.Error(), redpanda_service.ErrServiceNoLogFile.Error()) {
-			// Log the error but don't fail - this might happen during creation when the config file doesn't exist yet
-			// Note: as we use the logs of the underlying redpanda_monitor service, we need to ignore ErrServiceNoLogFile here.
-			b.baseFSMInstance.GetLogger().Debugf("Service not found, will be created during reconciliation: %v", getConfigErr)
-			return nil
-		}
-		if strings.Contains(getConfigErr.Error(), redpanda_monitor.ErrServiceConnectionRefused.Error()) {
-			// This is expected during the startup phase of the redpanda service, when the service is not yet ready to receive connections
-			b.baseFSMInstance.GetLogger().Debugf("Service not yet ready: %v", getConfigErr)
-			return nil
+	// Use a select statement to wait for either the g.Wait() result or the context's cancellation.
+	select {
+	case err := <-errc:
+		// g.Wait() has finished, so check if any goroutine returned an error.
+		if err != nil {
+			// If there was an error in any sub-call, return that error.
+			return err
 		}
-		return fmt.Errorf("failed to get observed Redpanda config: %w", getConfigErr)
+		// If err is nil, all goroutines completed successfully.
+	case <-ctx.Done():
+		// The context was canceled or its deadline was exceeded before all goroutines finished.
+		// Although some goroutines might still be running in the background,
+		// they use a context (gctx) that should cause them to terminate promptly.
+		// Experiments have shown that without using this, some goroutines can still take up to 70ms to terminate.
```

Comment:
maybe update the comment with the milliseconds

Reply:
Removed for now, as this is unknown for the occurences i have.